### DISCUSSION TITLE
Multitarget optimization for Formulations

### DISCUSSION THREAD

**Discussion by varys50** [emdgroup/baybe#381] [Q&A]
Hi.  I am trying to figure out if this is possible with BayBe.

I have a series of formulations containing some specific chemicals and some mixtures of unknown chemicals.  Imagine it's similar to the problem where you have some mixture of polymers and then some small molecules.  I know the weight percents of all the ingredients (small molecules and polymers).  I don't know the molar concentrations of the polymers.  There are several properties measured for these formulations.  I want to know if it is possible to predict the best possible formulations to test that would maximize the properties I am interested in.  The output would be weight percent of ingredients for the formulation (polymers + small molecules).

Is this possible?  Thanks in advance!

**Comment by Scienfitz** [comment]
Hi @varys50 

Based on your description I don't see anything out of the ordinary here, this problem seems to fit entirely in a relatively standard Bayesian optimization description.
- Parameters: weight percents of ingredients (it doesn't matter if they are single molecules or mixtures themselves)
- Constraints: Sum constraint for the total plus any other restriction you might have like min/max amounts etc
- Targets: the properties you're interested in
- Objective: Combine targets via [desirability](https://emdgroup.github.io/baybe/stable/userguide/objectives.html#desirabilityobjective), example [here](https://emdgroup.github.io/baybe/stable/examples/Multi_Target/desirability.html)

If you've already tried this let me know and perhaps provide more details or code snippets.

In BayBE you can treat mixtures with a [traditional approach](https://github.com/emdgroup/baybe/issues/365#issuecomment-2330858734) or a not so standard [slot-based approach](https://emdgroup.github.io/baybe/stable/examples/Constraints_Discrete/mixture_constraints.html).

**Comment by varys50** [comment]
Hi @Scienfitz ,

I tried to follow the traditional approach you mentioned so we can use that as an example.  How would I incorporate molecular features into the parameters along with the weight percents of these molecules?

**Reply by Scienfitz** [reply]
molecular features are exactly whats not possible with the traditional approach

thats possible with the slot-based approach, but that has other limitations

**Comment by varys50** [comment]
Thanks again for the quick responses.  So, in the slot-based approach, is there a need to explicitly weigh the Mordred descriptor values for each solvent by the Frac values as a separate parameter or is that implied somehow in that example?

**Reply by Scienfitz** [reply]
I dont thinks its necessary to make that explicit. If you provide the model the molecular features for each slot and the fraction each slot occupies, the model should be able to learn the connection.

I am also not a fan of simply weighing molecular features of the components comprising a mixture. The properties of the mixture are most often not simply a linear combination of the components features. Hence, I favor the slot based approach. However, its quite a bit more complicated to set up and will be computationally too demanding if you have many slots.

**Comment by Scienfitz** [comment]
closed as inactive
